File: backend/Services/Parsers/SquarewordParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class SquarewordParser : BasicScoreResultParser
    {
        private ILogger<SquarewordParser> _logger;

        public SquarewordParser(ILogger<SquarewordParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override string Category => "Word";
        public override string GameName => "Squareword";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"{GameName.ToLower()}.org \\d+: (?<{ScoreGroup}>\\d+) guesses");
        protected override string? ExtraContent => "#squareword #squareword121";
        public override string Url => "https://squareword.org/";

        protected override string GetCleanResult(string result, Match parserResults)
        {
            return base.GetCleanResult(result.Replace($"{GameName.ToLower()}.org", $"{GameName.ToLower()}"), parserResults);
        }
    }
}

File: lambdas/src/parsers/squareword-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class SquarewordParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Squareword';
  readonly helpText = null;
  protected readonly extraContent = '#squareword #squareword121';
  readonly url = 'https://squareword.org/';
  
  // Matches "squareword.org 1461: 6 guesses"
  protected readonly parser = new RegExp(
    `${this.gameName.toLowerCase()}.org \\d+: (?<${BasicScoreResultParser.SCORE_GROUP}>\\d+) guesses`
  );
  
  protected getCleanResult(result: string, match: RegExpMatchArray): string {
    const cleaned = result.replace(`${this.gameName.toLowerCase()}.org`, this.gameName.toLowerCase());
    return super.getCleanResult(cleaned, match);
  }
}
